### DISCUSSION TITLE
Histoire doesn't load tailwind styles

### DISCUSSION THREAD

**Discussion by swyxio** [histoire-dev/histoire#315] [Q&A] [answered]
### Describe the bug

see here: https://histoire-build-bug-demo.netlify.app/story/src-components-featurecard-story-svelte?variantId=_default

the repro/code for this is here: https://github.com/sw-yx/swyxkit/pull/113

this component is written in tailwind, it is supposed to look like this
![image](https://user-images.githubusercontent.com/6764957/194294838-f0a943f7-bbb2-4cbe-909e-bce03b37197d.png)

but histoire looks like this (both in dev and in build)
![image](https://user-images.githubusercontent.com/6764957/194295089-e1780104-cce7-4e7e-b1c4-4e13cf2f355a.png)

so i think its pretty clear that the tailwind styles are not loading.

### Reproduction

repro: https://github.com/sw-yx/swyxkit/pull/113 (in the same histoire branch, which i've overwritten to reflect the latest histoire version)

### System Info

```shell
System:
    OS: macOS 12.5.1
    CPU: (10) arm64 Apple M1 Max
    Memory: 197.92 MB / 32.00 GB
    Shell: 5.8.1 - /bin/zsh
  Binaries:
    Node: 16.15.0 - ~/Library/Caches/fnm_multishells/1002_1664767773366/bin/node
    Yarn: 1.22.19 - /opt/homebrew/bin/yarn
    npm: 8.5.5 - ~/Library/Caches/fnm_multishells/1002_1664767773366/bin/npm
  Browsers:
    Chrome: 105.0.5195.125
    Safari: 15.6.1
  npmPackages:
    @histoire/plugin-svelte: ^0.11.0 => 0.11.0
    histoire: ^0.11.0 => 0.11.0
```

**Comment by Akryum** [comment, marked as answer]
You need to add CSS with a [setup file](https://histoire.dev/guide/config.html#global-js-and-css).

Then in that setup file import the CSS file with:

```css
@tailwind base;
@tailwind components;
@tailwind utilities;
```

---

Change [the histoire config](https://github.com/sw-yx/swyxkit/blob/dd861e0a808ecf9734db226ef878f80a6aa5ff30/histoire.config.js#L7) with:

```ts
{
  // ...
  setupFile: '/src/histoire-setup.ts',
}
```

Create an `src/histoire-setup.ts` file:

```ts
import './tailwind.css'
```

**Reply by Akryum** [reply]
I added a section about it in the docs: https://histoire.dev/examples/tailwind.html

**Reply by swyxio** [reply]
thanks very much! 

my feedback from a DX perspective here is - it's odd for histoire to detect my tailwind config for design system tokens, but then NOT detect it for story implementation, and require extra config to make it work that is not documented in Getting Started. the [setupFile doc you linked](https://histoire.dev/guide/config.html#global-js-and-css) also does not mention tailwind at all even though tailwind users will have to figure it out like here. (aka it setupFile is optional for nontailwind users, mandatory for tailwind users)

seems like an inconsistent "promise" of tailwind support, you know what i mean?

like ideally i want Histoire to "support Tailwind out of the box", which is not exactly what is promised, but is what i assumed from browsing the docs... until i ran into this issue.

**Reply by Akryum** [reply]
I understand, but then how do we handle people that import their existing CSS which might already contain the tailwind directives? For now maybe it's better to not do some magic importing, however we could improve the DX with documentation, warnings or suggesting to generate some files.

**Reply by Akryum** [reply]
Maybe we could scan for CSS file and if 1. we find some and 2. you don't have a setup file yet, auto-generate a setup file which imports those.

**Reply by Akryum** [reply]
I also have a lot of ideas around DX in the frontend client that I will experiment on, like development-only menus and buttons. There is space to do things here.

**Reply by swyxio** [reply]
yeah improving the docs is a good first step, but the best docs is no docs :)

i like the idea of autogen setup file. 

anyway! im sure you have thought about this more than me :)

**Reply by Aragur** [reply]
@Akryum it's still not working for me using this guide…
https://histoire.dev/examples/tailwind.html
Has something changed?

**Reply by Ishadijcks** [reply]
Can confirm that it is still not working

**Reply by Ishadijcks** [reply]
It does seem to work after `build && preview`, just not in in `dev

**Reply by romanslonov** [reply]
Try clear your browser cache If tailwind's styles are not loading. For me it fixes issue.

**Reply by Ishadijcks** [reply]
Thanks for the tip. It was working for me suddenly yesterday, maybe that was the cause!

**Reply by Thebigbignooby** [reply]
@romanslonov thanks! One year later I am still running into this, your comment was helpful!

**Comment by Eliophot** [comment]
On the other hand, if you have dynamic colors, the rendering does not work unless you add this in the tailwind configuration: https://v2.tailwindcss.com/docs/optimizing-for-production#safelisting-specific-classes
